### PR TITLE
Fixes #11449: Allow variable as directive parameters in regexp validated inputs 

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/11449

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Close for now, this needs to be tested

File: rudder-core/src/main/scala/com/normation/cfclerk/domain/RegexConstraint.scala [hunk 1]
```diff
@@ -53,14 +53,14 @@ case class RegexConstraint(pattern: String, errorMsg: String) extends HashcodeCa
 
   val compiled = Pattern.compile(pattern)
 
+  val variablePattern = Pattern.compile(".*\\$\\{.*\\}.*")
```
Comment (ncharles) [inline rudder-core/src/main/scala/com/normation/cfclerk/domain/RegexConstraint.scala]: i fear it may be a bit too eager

Comment (ncharles) [reply rudder-core/src/main/scala/com/normation/cfclerk/domain/RegexConstraint.scala]: well, actually, i can't see why it should be an issue

**Comment by VinceMacBuche** [comment]
Commit modified

File: rudder-core/src/main/scala/com/normation/cfclerk/domain/RegexConstraint.scala (hunk 1 above)
Comment (VinceMacBuche) [reply rudder-core/src/main/scala/com/normation/cfclerk/domain/RegexConstraint.scala]: Yes we don't want to analyze it deeply do it does not need to be perfect matches for cases i tested ... (one var, two vars, var inside a var ... )

Comment (VinceMacBuche) [reply rudder-core/src/main/scala/com/normation/cfclerk/domain/RegexConstraint.scala]: I updated the regexp so it's less eager (added ? after * so it matches as less possible)

**Comment by VinceMacBuche** [comment]
Commit modified

**Pull request merged**